pkg/api/nextdate.go

```diff
@@ -0,0 +1,276 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// layout задаёт формат даты в виде строки "ГГГГММДД"
+const layout = "20060102"
+
+// afterNow возвращает true, если date позже времени now
+func afterNow(date, now time.Time) bool {
+	return date.After(now)
+}
+
+// convertToInts разбирает строку вида "1,2,3" в срез целых чисел
+func convertToInts(input string) ([]int, error) {
+	if input == "" {
+		return nil, errors.New("input string is empty")
+	}
+	// Делим строку на подстроки по запятой
+	parts := strings.Split(input, ",")
+	result := make([]int, 0, len(parts))
+	for _, part := range parts {
+		// Удаляем пробелы и конвертируем в число
+		num, err := strconv.Atoi(strings.TrimSpace(part))
+		if err != nil {
+			return nil, errors.New("failed to parse integer from input")
```

Comment:
теряешь сообщение с ошибкой
```suggestion
			return nil, fmt.Errof("failed to parse integer from input: %w",err)
```

Reply:
Исправила